### ISSUE TITLE
Move towards a progressive web app on the demo

### ISSUE THREAD

**Issue by tech4GT** [publiclab/image-sequencer#326]
We should add progressive elements to the demo, going with the recent trend. I have listed some basic elements below please feel free to suggest more 😄 
- [ ] manifest.json
- [ ] cache static assets for offline use
- [ ] Improve responsiveness of the design
- [ ] Move saving sequences to indexedDB API (possibly making it optional for user to store it using any api, currently this is done using localStorage API)

**Comment by jywarren** [comment]
love this!

**Comment by Mridul97** [comment]
I want to work on this issue. Can I work on this issue ?

**Comment by tech4GT** [comment]
Yes @Mridul97 this would be great, we need to finalize on some ideas though, maybe you can start with adding a manifest and caching static assets with a network first strategy and then we can where we can go from there. Thanks!

**Comment by Mridul97** [comment]
Sure! I will start with adding a manifest and caching static assets with a network first strategy.

**Comment by Mridul97** [comment]
What will be the theme colour for manifest.json file ?

**Comment by tech4GT** [comment]
Isn’t the dominant colour of the page automatically picked? I’m sorry if I don’t make sense here, I’m a little out of touch from this.

**Comment by tech4GT** [comment]
Oh i got it! You’re right can you use a color picker to get the exact colour of the `save sequence` button. Thanks!

**Comment by Mridul97** [comment]
Okay! will do that.

**Comment by Mridul97** [comment]
I have made a manifest.json file. Please see it, is it fine ?
![screenshot 35](https://user-images.githubusercontent.com/21158457/45159869-90e04e80-b205-11e8-802f-ba9710ea373a.png)
![screenshot 36](https://user-images.githubusercontent.com/21158457/45159870-90e04e80-b205-11e8-8dc3-b401a5264038.png)
![screenshot 37](https://user-images.githubusercontent.com/21158457/45159871-90e04e80-b205-11e8-8584-5a43834175a0.png)

**Comment by tech4GT** [comment]
This looks very promising @Mridul97 Way to go!! Thanks!

**Comment by Mridul97** [comment]
Thanks! @tech4GT :-)

**Comment by Mridul97** [comment]
Hey, could you please tell me to cache assets for offline use, first do I have to set up service worker or it has  to be done without using service worker?

**Comment by tech4GT** [comment]
We would need a service worker definitely or else the data will be flushed if the user clears browser cache. So Just cache all the assets like html, css, js, images an so on. Basically the idea here is to cache everything that is currently being served by the server, so that in case of network outage these files can be used. Please also try to flush these cache files if we get fresh data from the network. You can read more about this https://developers.google.com/web/ilt/pwa/caching-files-with-service-worker

**Comment by Mridul97** [comment]
Okay Thanks!

**Comment by Mridul97** [comment]
Hey, I have cache all the assets on install event of the service worker and as you said to use network first approach, the service worker will first send request to network and only if it fails then service worker will take response from the cache. Please see it. Is it fine ?
![screenshot 40](https://user-images.githubusercontent.com/21158457/45220812-17ac2e80-b2cd-11e8-944a-bd1732fa0a9d.png)
![screenshot 41](https://user-images.githubusercontent.com/21158457/45220813-1844c500-b2cd-11e8-8f99-ca97092a3437.png)
![screenshot 42](https://user-images.githubusercontent.com/21158457/45220814-1844c500-b2cd-11e8-9f22-e1b1928eeb1b.png)
![screenshot 43](https://user-images.githubusercontent.com/21158457/45220816-1844c500-b2cd-11e8-8804-58316548f72f.png)

**Comment by tech4GT** [comment]
This looks awesome @Mridul97 just one more thing, don't you think we should update the cache version if the files are changed? Just a thought.

**Comment by Mridul97** [comment]
Yes, we should update the cache version if the files are changed. I will do it. Thanks!

**Comment by Mridul97** [comment]
Hey, I have made changes to update the cache. When any of the files are changed, we also change the static cache name from something like 'image-sequencer-static-v1' to 'image-sequencer-static-v2'. This creates a new version of the service worker and we delete the old the caches on the activation of this new version of service worker. Please take a look.
![ss_2](https://user-images.githubusercontent.com/21158457/45254955-68d82300-b39e-11e8-97da-b859712393fe.jpg)
![ss_3](https://user-images.githubusercontent.com/21158457/45254956-68d82300-b39e-11e8-9963-df354254ae21.jpg)

**Comment by tech4GT** [comment]
This looks pretty good! One doubt though, the cache will update to V3 and so on right?

**Comment by Mridul97** [comment]
Yes, we will change the staticCacheName to 'image-sequencer-static-v3' and it will save this cache and delete all the other old caches.

**Comment by tech4GT** [comment]
This is great work!! Please open a pull request so that I can look at the diff. Thanks Mridul! Way to go!!

**Comment by Mridul97** [comment]
I will open the pull request. Thanks!

**Comment by tech4GT** [comment]
Actually we have to be careful with this stuff!! We don't want someone to be stuck with an old version of cache, especially someone with no technical knowledge. So please allow edits from maintainers. Thanks!

**Comment by Mridul97** [comment]
I have made a pull request #331. Please take a look. Thanks!
